codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -654,55 +654,49 @@ let actionReducer = (state: t, action: action) => {
-      let chainFetchers = chainManager.chainFetchers->ChainMap.map(cf => {
-        let {
-          chainConfig,
-          logger,
-          fetchState: {startBlock, endBlock, maxAddrInPartition},
-          dynamicContractPreRegistration,
-        } = cf
-
-        ChainFetcher.make(
-          ~dynamicContractRegistrations=dynamicContractPreRegistration->Option.mapWithDefault(
-            [],
-            Js.Dict.values,
-          ),
-          ~chainConfig,
-          ~lastBlockScannedHashes=ReorgDetection.LastBlockScannedHashes.empty(
-            ~confirmedBlockThreshold=chainConfig.confirmedBlockThreshold,
-          ),
-          ~staticContracts=chainConfig->ChainFetcher.getStaticContracts,
-          ~startBlock,
-          ~endBlock,
-          ~dbFirstEventBlockNumber=None,
-          ~latestProcessedBlock=None,
-          ~logger,
-          ~timestampCaughtUpToHeadOrEndblock=None,
-          ~numEventsProcessed=0,
-          ~numBatchesFetched=0,
-          ~processingFilters=None,
-          ~maxAddrInPartition,
-          ~dynamicContractPreRegistration=None,
-        )
-      })
+    Logging.info("Starting indexing after pre-registration")
+    let chainFetchers = chainManager.chainFetchers->ChainMap.map(cf => {
+      let {
+        chainConfig,
+        logger,
+        fetchState: {startBlock, endBlock, maxAddrInPartition},
+        dynamicContractPreRegistration,
+      } = cf
+
+      ChainFetcher.make(
+        ~dynamicContractRegistrations=dynamicContractPreRegistration->Option.mapWithDefault(
+          [],
+          Js.Dict.values,
+        ),
+        ~chainConfig,
+        ~lastBlockScannedHashes=ReorgDetection.LastBlockScannedHashes.empty(
+          ~confirmedBlockThreshold=chainConfig.confirmedBlockThreshold,
+        ),
+        ~staticContracts=chainConfig->ChainFetcher.getStaticContracts,
+        ~startBlock,
+        ~endBlock,
+        ~dbFirstEventBlockNumber=None,
+        ~latestProcessedBlock=None,
+        ~logger,
+        ~timestampCaughtUpToHeadOrEndblock=None,
+        ~numEventsProcessed=0,
+        ~numBatchesFetched=0,
+        ~processingFilters=None,
+        ~maxAddrInPartition,
+        ~dynamicContractPreRegistration=None,
+      )
+    })
 
-      let chainManager: ChainManager.t = {
-        chainFetchers,
-        arbitraryEventQueue: [],
-        isInReorgThreshold: false,
-        isUnorderedMultichainMode: chainManager.isUnorderedMultichainMode,
-      }
+    let chainManager: ChainManager.t = {
+      chainFetchers,
+      arbitraryEventQueue: [],
+      isInReorgThreshold: false,
+      isUnorderedMultichainMode: chainManager.isUnorderedMultichainMode,
+    }
 
-      let freshState = make(~config, ~chainManager, ~loadLayer)
+    let freshState = make(~config, ~chainManager, ~loadLayer)
 
-      (freshState, [NextQuery(CheckAllChains)])
-    } else {
-      (state, [])
-    }
+    (freshState->incrementId, [NextQuery(CheckAllChains)])
```

Comment:
Here the new state increments the state id so that any actions being dispatched from previous invalid state will be invalidated.